ultrasonic/src/main/java/org/moire/ultrasonic/util/EntryByDiscAndTrackComparator.java

```diff
@@ -33,7 +35,14 @@ public int compare(MusicDirectory.Entry x, MusicDirectory.Entry y)
 			return discComparison;
 		}
 
-		return compare(trackX == null ? 0 : trackX, trackY == null ? 0 : trackY);
+		int trackComparison = compare(trackX == null ? 0 : trackX, trackY == null ? 0 : trackY);
+
+		if (trackComparison != 0)
+		{
+			return trackComparison;
+		}
+
+		return compare(pathX == null ? "" : pathX, pathY == null ? "" : pathY);
```

Comment:
I have a doubt with this. `compare` is a function that compares two _longs_, but `pathX` and `pathY` are strings. I think that this is a `TypeError`

Reply:
It's overloaded, there is a `private static int compare(String a, String b)` on line 53